Mathlib/Util/GetAllModules.lean

```diff
@@ -0,0 +1,72 @@
+/-
+Copyright (c) 2024 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, Kim Morrison, Damiano Testa
+-/
+
+import Lean.Util.Path
+import Lake.CLI.Main
+
+
+/-!
+# Utility functions for finding all `.lean` files or modules in a project.
+
+TODO:
+`getLeanLibs` contains a hard-coded choice of which dependencies should be built and which ones
+should not.  Could this be made more structural and robust, possibly with extra `Lake` support?
+
+-/
+
+open Lean System.FilePath
+
+/-- `getAllFiles git ml` takes all `.lean` files in the directory `ml`
+(recursing into sub-directories) and returns the `Array` of `String`s
+```
+#[file₁, ..., fileₙ]
+```
+of all their file names.
+
+The input `git` is a `Bool`ean flag:
+* `true` means that the command uses `git ls-files` to find the relevant files;
+* `false` means that the command recursively scans all dirs searching for `.lean` files.
+-/
+def getAllFiles (git : Bool) (ml : String) : IO (Array System.FilePath) := do
+  let ml.lean := addExtension ⟨ml⟩ "lean"  -- for example, `Mathlib.lean`
+  let allModules : Array System.FilePath ← (do
+    if git then
+      let mlDir := ml.push pathSeparator   -- for example, `Mathlib/`
+      let allLean ← IO.Process.run { cmd := "git", args := #["ls-files", mlDir ++ "*.lean"] }
+      return (((allLean.dropRightWhile (· == '\n')).splitOn "\n").map (⟨·⟩)).toArray
+    else do
+      let all ← walkDir ml
+      return all.filter (·.extension == some "lean"))
+  let files := (allModules.erase ml.lean).qsort (·.toString < ·.toString)
+  let withImport ← files.mapM fun f => do
```

Comment:
```suggestion
  let withImport ← files.mapM fun f => do
```
The entries in the array no longer have `import` in them, so maybe it can be renamed to `existingFiles` or something similar?

Reply:
Good point, changed (and tested locally).